apps/api/src/users/users.service.ts

```diff
@@ -132,4 +132,8 @@ export class UsersService {
     );
     return this.authService.login(user);
   }
+
+  async updateName(user: User, new_username: string): Promise<UpdateResult> {
+    return this.usersRepository.update({ id: user.id }, { name: new_username });
+  }
```

Comment:
Not sure if I get this right, but it feels like you are not checking if the username already exists in the database. I would suggest: 
```suggestion
  async updateName(user: User, new_username: string): Promise<UpdateResult | null > {
      const usernameTaken = await this.usersRepository
        .createQueryBuilder()
        .where('LOWER(name) = :name', {
          name: new_username.toLowerCase(),
        })
        .getOne()
    if (usernameTaken)
        return null;
    return this.usersRepository.update({ id: user.id }, { name: new_username });
  }
```
(null as a return in case of failure is really just a suggestion btw, it might not be the most appropriate value in you case)

Reply:
Right. I have used the `UsersService.userExists` method here, and thrown an exception in case the username is already in use.